commit 92ce25dd8a45ba65367aba68434a461544a8eb60 (refs/changes/79/1079/1)
Author: Suk Won Choi <sw00.choi@samsung.com>
Date:   2016-12-07T11:21:35+09:00 (2 years, 10 months ago)
    
    joyent/sdc-imgapi#11 no_proxy support; https://cr.joyent.us/#/c/830/ commit should be applied to package.json

diff --git a/lib/common.js b/lib/common.js
index 5ab60e1..c7c7c82 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -742,6 +742,11 @@ function httpClientOpts(clientOpts, req) {
         clientOpts.proxy = app.config.httpProxy || false;
     }
 
+    // noProxy
+    if (clientOpts.noProxy === undefined) {
+        clientOpts.noProxy = app.config.noProxy;
+    }
+
     // headers.request-id
     if (req && req.getId()) {
         var req_id = (clientOpts.headers
diff --git a/sapi_manifests/docker/template b/sapi_manifests/docker/template
index 1bfa4b0..922f50e 100644
--- a/sapi_manifests/docker/template
+++ b/sapi_manifests/docker/template
@@ -59,6 +59,7 @@
     },
 
     "httpProxy": "{{{http_proxy}}}",
+    "noProxy": "{{{no_proxy}}}",
     "dockerRegistryInsecure": {{^docker_registry_insecure}}false{{/docker_registry_insecure}}{{#docker_registry_insecure}}{{{docker_registry_insecure}}}{{/docker_registry_insecure}},
 
     "useTls": {{^USE_TLS}}false{{/USE_TLS}}{{#USE_TLS}}{{{USE_TLS}}}{{/USE_TLS}},
